### PR TITLE
Delete MLIR-AIE submodule

### PR DESCRIPTION
**Pull request by makslevental**
This PR delete/removes the MLIR-AIE submodule. The load-bearing change is forking all of the `AIEOps.td` type things. The main reason for doing this now is https://github.com/Xilinx/mlir-aie/pull/1660, which does not fix the parser issue that broke us _but makes it permanent_.

In performing this fork we take the most minimal set of auxiliary/support impls/function/code. **Specifically, we take none of their verifiers**. The reason is, as usual, 1) we don't intend to continue to have most/many of these ops 2) they're now not an egress boundary i.e., they're purely "compiler state" ops/IR. Naturally, whichever ops we keep or iterate on, we should have verifiers for but I prefer 1) to add them incrementally rather than eagerly 2) add them in a separate PR (or PRs).

A couple things to call out:

1. The tests do not change (modulo `i32` -> `i8`). This is an indication that the fork/port/vendoring is successful;
    * This change, which brings in closer alignment with `aie-rt`, incurs that cost that `uint8_t`s need to printed with `std::to_string`[^3];
3. The prior vendored passes incur some minor changes because in doing the fork I removed all superfluous/redundant/confusing getters[^1];
4. Rather than adding patches to https://github.com/nod-ai/mlir-air I add a few local `sed`s. I did it this way because the patches would break an isolated build there[^2]. Not the prettiest thing but I'll keep up with it.

~~Note, there's a still bit to do (I want to rename the `xilinx::AIE` namespace and then use `typedef`s) but this is ready to review.~~

On second thought let's merge this and then move all the ops/stuff to our `mlir::iree_compiler::AMDAIE` namespace in a follow-up.

Fixes https://github.com/nod-ai/iree-amd-aie/issues/430

[^1]: Stuff like `getIDInt` when the existing attr getter is `getID`.
[^2]: Which never happens sure but just seems like it sprawls the hackiness rather than containing it to one place.
[^3]: Otherwise you print the ASCII character at the position rather the number itself.

### CODE REVIEW

**Review by jtuyls — APPROVED** [review]
LGTM

**Comment by makslevental** [comment]
@nirvedhmeshram @newling @Abhishek-Varma you guys want to take a look or you're good with this?

File: build_tools/ci/build_test_cpp.sh [hunk 1]
```diff
@@ -7,6 +7,7 @@ repo_root="$(cd $this_dir/../.. && pwd)"
 iree_dir="$(cd $repo_root/../iree && pwd)"
 build_dir="$repo_root/iree-build"
 install_dir="$repo_root/iree-install"
+rm -rf "$build_dir" "$install_dir"
```
Comment (nirvedhmeshram) [inline build_tools/ci/build_test_cpp.sh:10]: Is this needed? (just thinking of some poor user who uses it and then deletes everything on what should have been an incremental change)

**Review by nirvedhmeshram — APPROVED** [review]
LGTM, just a minor question but you can address however and land.

File: build_tools/ci/build_test_cpp.sh (hunk 1 above)
Comment (makslevental) [reply build_tools/ci/build_test_cpp.sh:10]: Nah this was just to debug why CI was failing when local was passing. I'll remove

**Pull request merged**